src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -79,19 +79,35 @@ public void startTask(final String tableName, final String topic,
   }
 
   @Override
-  public void insert(final SinkRecord record)
+  public void insert(final Collection<SinkRecord> records)
```

Comment:
This function will be triggered periodically with empty collection. Every time that happens, we will check whether we should flush a pipe based on current time and the last time it was flushed.